### PR TITLE
Update oldest deps for `pyarrow` & `numpy`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
We recently pinned our `dask-expr` version to `1.1.14`: https://github.com/rapidsai/rapids-dask-dependency/pull/64, that plus latest `dask` seems to be having a minimum requirement for `pyarrow` as `14.0.1`. This is causing failures in our CI matrix while running tests with the oldest dependencies. This PR bumps the minimum pyarrow version in our oldest deps.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by galipremsagar** [comment]
> It looks like the current installation in the conda testing scripts is too broad. Let's change test_python_common to accept the file_key as an argument that we can pass [here](https://github.com/rapidsai/cudf/blob/branch-24.10/ci/test_python_common.sh#L16), then make separate groups for test_python_cudf and test_python_other.

Done 👍

File: ci/test_python_cudf.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@
 cd "$(dirname "$(realpath "${BASH_SOURCE[0]}")")"/../;
 
 # Common setup steps shared by Python test jobs
-source ./ci/test_python_common.sh
+source ./ci/test_python_common.sh test_python_cudf
```
Comment (vyasr) [inline ci/test_python_cudf.sh:8]: You'll need to make a key corresponding to this, or just rename the test_python key and remove the dask_cudf and cudf_pandas lists from that key.

Comment (galipremsagar) [reply ci/test_python_cudf.sh:8]: Done, could you take a look at the new changes? I had to retain `test_python` as that was being utilized by `ci/cudf_pandas_scripts/run_tests.sh`

Comment (vyasr) [reply ci/test_python_cudf.sh:8]: I would rename that to `test_python_cudf_pandas` and remove the unnecessary bits like polars to minimize the number of packages being installed into that environment.

Comment (galipremsagar) [reply ci/test_python_cudf.sh:8]: Is it okay if I push this particular change in a follow-up in the interest of unblocking the CI ASAP?

Comment (vyasr) [reply ci/test_python_cudf.sh:8]: Sure that is fine.

Comment (galipremsagar) [reply ci/test_python_cudf.sh:8]: CI is now passing and I incorporated the changes discussed here into this PR.

**Review by vyasr — APPROVED** [review]

**Comment by vyasr** [comment]
Thanks Prem!

**Comment by galipremsagar** [comment]
> Thanks Prem!

Thank you for helping me in getting this out!

**Pull request merged**